R/test-files.r

```diff
@@ -155,7 +155,15 @@ test_file <- function(path, reporter = "summary", env = test_env(),
   fname <- basename(path)
   lister$start_file(fname)
 
-  sys.source2(fname, new.env(parent = env))
+  tryCatch({
+    sys.source2(fname, new.env(parent = env))
+  }, error = function(e){
+    test_that(
```

Comment:
I like the idea, but I'm not sure this will work correctly. I think it should be more like:

```R
context(fname)
fail("Parsing failed")
```